### PR TITLE
dzsave.c: Remove trailing comma from IIIF folder names

### LINKED ISSUE #2589: IIIFv3 folder names generated by dzsave contain trailing comma
**Describe the bug**
If you use the command `vips dzsave --layout iiif3 example.png example` to convert an image to an IIIFv3 folder structure, the resulting folders on the second level will all have a trailing comma in their name, e.g. `/example/1536,1536,512,464/512,464,/0/default.jpg`. All other software in the IIIF ecosystem that I've encountered so far, such as [OpenSeadragon](https://openseadragon.github.io/) or [Dezoomify](https://github.com/lovasoa/dezoomify-rs), don't expect this trailing comma, so requests end up returning a `404 - Not Found` error.

**To Reproduce**
1. Run the command `vips dzsave --layout iiif3 example.png example` in a folder with an image named `example.png`.
2. Browse to the second level of the newly created `example` folder and check that the folder names contain a trailing comma.

**Expected behavior**
Generated folder names don't contain a trailing comma.

**Actual behavior**
Generated folder names do contain a trailing comma.

**Screenshots**
N/A

**Environment**
 - **OS:** Windows 10 & Debian 11
 - **Vips:** v8.12.1

**Additional context**
N/A

### PR DESCRIPTION
**Pull request by whalehub**
Fixes #2589.

### CODE REVIEW

File: libvips/foreign/dzsave.c [hunk 1]
```diff
@@ -1495,7 +1497,7 @@ tile_name( Layer *layer, int x, int y )
 			int size = VIPS_MIN( dz->tile_size, 
 				layer->width - x * dz->tile_size );
 
-			vips_snprintf( dirname2, VIPS_PATH_MAX, "%d,", size );
+			vips_snprintf( dirname2, VIPS_PATH_MAX, "%d", size );
```
Comment (jcupitt) [inline libvips/foreign/dzsave.c:1500]: I believe the trailing comma is needed for iiif2 -- I think it means "ysize == xsize" (I think).

Comment (whalehub) [reply libvips/foreign/dzsave.c:1500]: @jcupitt Just had another look and it turns out you're right, it is needed for IIIFv2. I'll amend the PR.

**Comment by whalehub** [comment]
@jcupitt It should be good to merge now. 👍

**Comment by jcupitt** [comment]
That's great @whalehub, thank you for reporting and fixing this error! I'll apply the fix to 8.12 too.

**Pull request merged**